### PR TITLE
Merge pull request #1751 from r-glyde/file-permissions

### PR DESCRIPTION
**Pull request by r-glyde**
Part of #1533 allowing getting and setting file permissions

### CODE REVIEW

File: io/src/test/scala/fs2/io/file/FileSpec.scala [hunk 1]
```diff
@@ -127,6 +130,59 @@ class FileSpec extends BaseFileSpec {
...
+        .unsafeRunSync()
+        .isLeft shouldBe true
+    }
+    "should return permissions for existing file" in {
+      val permissions = PosixFilePermissions.fromString("rwxrwxr-x")
+      Blocker[IO]
+        .use { b =>
+          tempFile
+            .evalMap { p =>
+              IO.delay(Files.setPosixFilePermissions(p, permissions)) >> file.permissions[IO](b, p)
```
Comment (kubukoz) [inline io/src/test/scala/fs2/io/file/FileSpec.scala]: I think it's fine to use the fs2 method for setting permissions - that way we'll verify that the methods interact well with each other. I'd wait for someone else to state their opinion on this, though.

Comment (r-glyde) [reply io/src/test/scala/fs2/io/file/FileSpec.scala]: Yeah, I think that should be fine as well. It felt a bit odd when I first did it that way but will just be the same in the end and a bit tidier as well. I'll wait for someone to confirm before pushing a change.

Comment (mpilquist) [reply io/src/test/scala/fs2/io/file/FileSpec.scala]: Yeah I think it's fine to use fs2 method here

Comment (r-glyde) [reply io/src/test/scala/fs2/io/file/FileSpec.scala]: Updated to just use the fs2 methods in both tests

**Comment by kubukoz** [comment]
Thanks!

**Pull request merged**